### PR TITLE
Only update hover execution state if cell is not active.

### LINKED ISSUE #18980: When prompting for user input in Interactive cell, the input window closes immediately after opening
Run this code in a C# cell to try and add user input. The input field closes immediately, so it might be an issue with something else stealing the window focus.

```
var input = await GetInputAsync("give me data");
input
```

### PR DESCRIPTION
**Pull request by corivera**
Currently, prompting for user input with .NET Interactive breaks when mousing over the cell because we always re-focus on the cell when changing the execution state on hover. So if you click run on a cell that prompts for input and mouse anywhere off of the run button, then the input box closes immediately. The hover state is only used to determine if the cell's Run button should be visible, so my fix here is to only fire the hover state change if the cell isn't active, since the Run button is always visible while active anyway.

This PR fixes #18980

### CODE REVIEW

File: src/sql/workbench/services/notebook/browser/models/cell.ts [hunk 1]
```diff
@@ -287,7 +287,10 @@ export class CellModel extends Disposable implements ICellModel {
 
 	public set hover(value: boolean) {
 		this._hover = value;
-		this.fireExecutionStateChanged();
+		// Skip changing the execution state while running to prevent focus changes from breaking things like user input
+		if (!this.isRunning) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/notebook/browser/models/cell.ts:291]: Did you try removing this event firing completely? I'm not sure why we would ever need to fire the event for this - the state isn't actually changing.

Comment (corivera) [reply src/sql/workbench/services/notebook/browser/models/cell.ts:291]: The execution state is used to control the class state of the RunCell button: https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsActions.ts#L121

The state gets set from the executionState event here: https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViewsActions.ts#L153

If you skip firing the event, then the Run button doesn't show up when hovering over an inactive cell.

Comment (Charles-Gagnon) [reply src/sql/workbench/services/notebook/browser/models/cell.ts:291]: Sigh. This really shouldn't be named "cell execution state". Having the cell execution state be "hidden" makes no sense. 

Welp, probably out of scope for this. But it'd be great to look into at least updating the name of this thing so it's clearer that it's more than just the current state of the cell execution. Maybe `CellExecutionIconState`? Not sure if this is used for anything else than the icon though.

**Review by Charles-Gagnon — APPROVED** [review]

**Review by MaddyDev — APPROVED** [review]

File: src/sql/workbench/services/notebook/browser/models/cell.ts (hunk 1 above)
Comment (corivera) [reply src/sql/workbench/services/notebook/browser/models/cell.ts]: Looks like CellExecutionState used in a few other places, so it isn't just for the icon state.

Comment (Charles-Gagnon) [reply src/sql/workbench/services/notebook/browser/models/cell.ts]: 😭

**Comment by Charles-Gagnon** [comment]
If you plan on bringing this to ask mode don't link it to auto-close here. We want it to stay open until it's ported over to the release branch.

**Comment by corivera** [comment]
CI failures are from the manage packages test again, which is unrelated to these changes. Going ahead with the merge.

**Pull request merged**